samples/OpenIDConnectSample/Startup.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.AspNet.Security.OpenIdConnect;
+using Microsoft.AspNet.Http.Security;
+using Microsoft.AspNet.Security;
+
+namespace OpenIdConnectSample
+{
+    public class Startup
+    {
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseServices(services =>
+            {
+                services.AddDataProtection();
+                services.Configure<ExternalAuthenticationOptions>(options =>
+                {
+                    options.SignInAsAuthenticationType = OpenIdConnectAuthenticationDefaults.AuthenticationType;
+                });
+
+            });
+
+            app.UseCookieAuthentication(options =>
+            {
+                options.AuthenticationType = OpenIdConnectAuthenticationDefaults.AuthenticationType;
+            });
+
+            app.UseOpenIdConnectAuthentication(options =>
+                {
+                    options.ClientId = "fe78e0b4-6fe7-47e6-812c-fb75cee266a4";
+                    options.Authority = "https://login.windows.net/cyrano.onmicrosoft.com";
+                    options.RedirectUri = "http://localhost:42023";
+                    options.SignInAsAuthenticationType = OpenIdConnectAuthenticationDefaults.AuthenticationType;
```

Comment:
I think it's time to have a check in all the providers that use a `SignInAsAuthenticationType` property to make sure that it can't be that same as the `AuthenticationType`: an authentication handler shouldn't call itself to delegate the persistence of the authentication ticket. This kind of cyclic call is definitely not a valid scenario.

@Tratcher your thoughts?